### PR TITLE
Adding test cases for DS-1186

### PR DESCRIPTION
**Pull request by anupama-pathirage**
Added test cases for DS-1186-JSON message rendering fails in DSS 3.2.2 with &. Test cases are added for 3.5.0 version.

### CODE REVIEW

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+    private void deleteResource()
+            throws RemoteException, MalformedURLException, ResourceAdminServiceExceptionException,
+            XPathExpressionException {
+        ResourceAdminServiceClient resourceAdmin = new ResourceAdminServiceClient(dssContext.getContextUrls().getBackEndUrl()
+                , sessionCookie);
+        resourceAdmin.deleteResource("/_system/config/automation/resources/policies/");
+    }
+
```
Comment (chanikag) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Formatting issue: unnecessary spaces.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    public void destroy() throws Exception {
+        deleteService(serviceName);
+        deleteResource();
+        cleanup();
+    }
+
+    @Test(groups = {"wso2.dss"}, description = "Check whether the JSON render service works when there is '&'", alwaysRun = true)
+    public void jsonRenderWithSecurity() throws Exception {
+        HttpResponse response = this.getHttpResponse(serviceEndPoint + "status", "application/json");
+        String receivedResult=response.getData();
```
Comment (Manuri) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Need to add spaces near =.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            String encode = (new String((new Base64()).encode((userInfo.getUserName() + ":" + userInfo.getPassword())
+                        .getBytes()))).replaceAll("\n", "");
+                conn.setRequestProperty("Authorization", "Basic " + encode);
+
+            conn.setReadTimeout(10000);
+            conn.connect();
+            // Get the response
+            StringBuilder sb = new StringBuilder();
+            BufferedReader rd = null;
+            try {
```
Comment (Manuri) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: I suppose we can use try with resources[1] here.

[1] https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    public void jsonRenderWithSecurity() throws Exception {
+        HttpResponse response = this.getHttpResponse(serviceEndPoint + "status", "application/json");
+        String receivedResult=response.getData();
+        String expectedResult = "{\"Entries\":{\"Entry\":[{\"status\":\"1 & 2\"}]}}";
+        Assert.assertNotNull(receivedResult, "Response is null");
+        Assert.assertTrue(expectedResult.equals(receivedResult.toString()), "Expected result not found");
+    }
+
+    /**
+     * This method will "Accept" header Types "application/json", etc..
```
Comment (Manuri) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Java doc best practices[1] include having a new line in between the method description and block tags

[1] http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            URL url = new URL(urlStr);
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("GET");
+            conn.setDoOutput(true);
+            conn.setRequestProperty("Accept", contentType);
+            conn.setRequestProperty("charset", "UTF-8");
+
+            String encode = (new String((new Base64()).encode((userInfo.getUserName() + ":" + userInfo.getPassword())
+                        .getBytes()))).replaceAll("\n", "");
+                conn.setRequestProperty("Authorization", "Basic " + encode);
```
Comment (lahirus) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Minor indentation issue.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java (hunk 3 above)
Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Fixed with PR: #97

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java (hunk 4 above)
Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Fixed with PR: #97

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java (hunk 5 above)
Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1186JsonRenderTestCase.java]: Fixed with PR: #97

**Pull request merged**